From 45ead30aeba3e279305f6b24bef7b66f94962afd Mon Sep 17 00:00:00 2001
From: Patrick Yavitz <pyavitz@xxxxx.com>
Date: Tue, 25 Jul 2023 06:55:34 -0400
Subject: [PATCH] arch: arm64: dts: amlogic: meson-sm1-x96-air: overlays

meson-g12-gxl-cma-pool-896MB.dtbo
meson-sm1-x96-air-rtc.dtbo

Signed-off-by: Patrick Yavitz <pyavitz@xxxxx.com>
---
 Makefile                                      |  2 +-
 arch/arm64/boot/.gitignore                    |  2 ++
 arch/arm64/boot/dts/amlogic/Makefile          |  2 ++
 arch/arm64/boot/dts/amlogic/overlays/Makefile |  7 +++++++
 .../overlays/meson-g12-gxl-cma-pool-896MB.dts | 19 +++++++++++++++++++
 .../overlays/meson-sm1-x96-air-rtc.dts        | 11 +++++++++++
 scripts/Makefile.dtbinst                      |  3 ++-
 scripts/Makefile.lib                          | 13 +++++++++++++
 8 files changed, 57 insertions(+), 2 deletions(-)
 create mode 100644 arch/arm64/boot/dts/amlogic/overlays/Makefile
 create mode 100644 arch/arm64/boot/dts/amlogic/overlays/meson-g12-gxl-cma-pool-896MB.dts
 create mode 100644 arch/arm64/boot/dts/amlogic/overlays/meson-sm1-x96-air-rtc.dts

diff --git a/Makefile b/Makefile
index 6c940cce3656..7b4c8f94e5ac 100644
--- a/Makefile
+++ b/Makefile
@@ -1471,7 +1471,7 @@ ifneq ($(dtstree),)
 	$(Q)$(MAKE) $(build)=$(dtstree) $(dtstree)/$@
 
 %.dtbo: dtbs_prepare
-	$(Q)$(MAKE) $(build)=$(dtstree) $(dtstree)/$@
+	$(Q)$(MAKE) $(build)=$(dtstree)/overlays $(dtstree)/overlays/$@
 
 PHONY += dtbs dtbs_prepare dtbs_install dtbs_check
 dtbs: dtbs_prepare
diff --git a/arch/arm64/boot/.gitignore b/arch/arm64/boot/.gitignore
index af5dc61f8b43..a5ce68fdfa7f 100644
--- a/arch/arm64/boot/.gitignore
+++ b/arch/arm64/boot/.gitignore
@@ -2,3 +2,5 @@
 Image
 Image.gz
 vmlinuz*
+*.dtb*
+*.scr
diff --git a/arch/arm64/boot/dts/amlogic/Makefile b/arch/arm64/boot/dts/amlogic/Makefile
index e213aeebb774..7af675846c13 100644
--- a/arch/arm64/boot/dts/amlogic/Makefile
+++ b/arch/arm64/boot/dts/amlogic/Makefile
@@ -1,4 +1,6 @@
 # SPDX-License-Identifier: GPL-2.0
+subdir-y += overlays
+
 dtb-$(CONFIG_ARCH_MESON) += meson-a1-ad401.dtb
 dtb-$(CONFIG_ARCH_MESON) += meson-axg-jethome-jethub-j100.dtb
 dtb-$(CONFIG_ARCH_MESON) += meson-axg-jethome-jethub-j110-rev-2.dtb
diff --git a/arch/arm64/boot/dts/amlogic/overlays/Makefile b/arch/arm64/boot/dts/amlogic/overlays/Makefile
new file mode 100644
index 000000000000..f4ea22c9f5a9
--- /dev/null
+++ b/arch/arm64/boot/dts/amlogic/overlays/Makefile
@@ -0,0 +1,7 @@
+# SPDX-License-Identifier: GPL-2.0
+dtbo-y += \
+	meson-g12-gxl-cma-pool-896MB.dtbo \
+	meson-sm1-x96-air-rtc.dtbo
+
+targets += $(dtbo-y)
+always-y := $(dtbo-y)
diff --git a/arch/arm64/boot/dts/amlogic/overlays/meson-g12-gxl-cma-pool-896MB.dts b/arch/arm64/boot/dts/amlogic/overlays/meson-g12-gxl-cma-pool-896MB.dts
new file mode 100644
index 000000000000..f8c476b04e8c
--- /dev/null
+++ b/arch/arm64/boot/dts/amlogic/overlays/meson-g12-gxl-cma-pool-896MB.dts
@@ -0,0 +1,19 @@
+/dts-v1/;
+/plugin/;
+
+/ {
+	compatible = "amlogic,g12a", "amlogic,g12b", "amlogic,meson-gxl";
+
+	fragment@0 {
+		target-path = "/reserved-memory";
+		__overlay__ {
+			linux,cma {
+				compatible = "shared-dma-pool";
+				reusable;
+				size = <0x0 0x38000000>;
+				alignment = <0x0 0x400000>;
+				linux,cma-default;
+			};
+		};
+	};
+};
diff --git a/arch/arm64/boot/dts/amlogic/overlays/meson-sm1-x96-air-rtc.dts b/arch/arm64/boot/dts/amlogic/overlays/meson-sm1-x96-air-rtc.dts
new file mode 100644
index 000000000000..e803198b5bc5
--- /dev/null
+++ b/arch/arm64/boot/dts/amlogic/overlays/meson-sm1-x96-air-rtc.dts
@@ -0,0 +1,11 @@
+/dts-v1/;
+/plugin/;
+
+/ {
+	fragment@0 {
+		target = <&i2c1>;
+		__overlay__ {
+			status = "okay";
+		};
+	};
+};
diff --git a/scripts/Makefile.dtbinst b/scripts/Makefile.dtbinst
index 190d781e84f4..84c46c081218 100644
--- a/scripts/Makefile.dtbinst
+++ b/scripts/Makefile.dtbinst
@@ -18,9 +18,10 @@ include $(srctree)/scripts/Kbuild.include
 include $(src)/Makefile
 
 dtbs    := $(addprefix $(dst)/, $(dtb-y) $(if $(CONFIG_OF_ALL_DTBS),$(dtb-)))
+dtbos   := $(addprefix $(dst)/, $(dtbo-y) $(if $(CONFIG_OF_ALL_DTBS),$(dtb-)))
 subdirs := $(addprefix $(obj)/, $(subdir-y) $(subdir-m))
 
-__dtbs_install: $(dtbs) $(subdirs)
+__dtbs_install: $(dtbs) $(dtbos) $(subdirs)
 	@:
 
 quiet_cmd_dtb_install = INSTALL $@
diff --git a/scripts/Makefile.lib b/scripts/Makefile.lib
index 3aa384cec76b..ecae974bd58c 100644
--- a/scripts/Makefile.lib
+++ b/scripts/Makefile.lib
@@ -339,6 +339,7 @@ DTC_FLAGS += -Wno-interrupt_provider
 # Disable noisy checks by default
 ifeq ($(findstring 1,$(KBUILD_EXTRA_WARN)),)
 DTC_FLAGS += -Wno-unit_address_vs_reg \
+	-Wno-gpios_property \
 	-Wno-avoid_unnecessary_addr_size \
 	-Wno-alias_paths \
 	-Wno-graph_child_address \
@@ -408,6 +409,18 @@ $(obj)/%.dtb: $(src)/%.dts $(DTC) $(DT_TMP_SCHEMA) FORCE
 $(obj)/%.dtbo: $(src)/%.dts $(DTC) FORCE
 	$(call if_changed_dep,dtc)
 
+quiet_cmd_dtco = DTCO    $@
+cmd_dtco = mkdir -p $(dir ${dtc-tmp}) ; \
+	$(CPP) $(dtc_cpp_flags) -x assembler-with-cpp -o $(dtc-tmp) $< ; \
+	$(DTC) -@ -H epapr -O dtb -o $@ -b 0 \
+		-i $(dir $<) $(DTC_FLAGS) \
+		-Wno-interrupts_property \
+		-d $(depfile).dtc.tmp $(dtc-tmp) ; \
+	cat $(depfile).pre.tmp $(depfile).dtc.tmp > $(depfile)
+
+$(obj)/%.dtbo: $(obj)/%.dts FORCE
+	$(call if_changed_dep,dtco)
+
 dtc-tmp = $(subst $(comma),_,$(dot-target).dts.tmp)
 
 # Bzip2
-- 
2.39.2

